### ISSUE TITLE
building an app with 'cmake in tree' is broken when using CUDA target

### ISSUE THREAD

**Issue by pkestene** [kokkos/kokkos#2469]
Hi,

### How to reproduce (from latest develop branch): 

Just try to configure example _build_cmake_in_tree_ (from kokkos examples) for a CUDA target:
```shell
export CXX=${KOKKOS_SRC}/bin/nvcc_wrapper
ccmake -DKokkos_ENABLE_CUDA=ON -DKokkos_CXX_STANDARD=11 -DKokkos_ENABLE_CUDA_LAMBDA=ON -DKokkos_ARCH_KEPLER30=ON ..
```

The last lines of **cmake output** are
```shell
-- The project name is: Kokkos
-- Using -std=c++11 for C++11 standard as feature
CMake Error at ../../cmake/kokkos_test_cxx_std.cmake:39 (MESSAGE):
  Compiler NVIDIA should support cxx_std_11, but CMake reports feature not
  supported
Call Stack (most recent call first):
  ../../cmake/kokkos_test_cxx_std.cmake:54 (kokkos_set_cxx_standard_feature)
  ../../cmake/kokkos_tribits.cmake:182 (INCLUDE)
  ../../CMakeLists.txt:114 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
```
The first commit where it was **not ok** is 0ad01d7c7907d4eead2f9ba65bbb42c3a3d4965e

The last commit where it was **working ok** is 7b14c3d8311194ed6c0a45a3a003b7728b1d3d0f  

The last lines of **cmake output** where it was working are:
```shell
-- The project name is: Kokkos
-- Using -std=c++11 for C++11 standard as feature
-- nvcc_wrapper does not support TARGET_COMPILE_FEATURES
-- Performing Test CXX_STD_FLAGS_ACCEPTED
-- Performing Test CXX_STD_FLAGS_ACCEPTED - Success
-- Compiler features not supported, but NVIDIA accepts -std=c++11
-- Found LIBDL: /usr/lib/x86_64-linux-gnu/libdl.so  
-- Execution Spaces:
--   Device Parallel: CUDA
--     Host Parallel: NONE
--       Host Serial: SERIAL
-- 
-- Architectures:
--  KEPLER30
-- Configuring done
-- Generating done

```

The **problem** is located in file cmake/kokkos_test_cxx_std.cmake, at the end of function kokkos_set_cxx_standard_feature, indeed CMAKE_CXX_COMPILE_FEATURES is empty for NVIDIA, so it generates a fatal error.

The last commit where is was working, the code function was checking for KOKKOS_CXX_COMPILER_ID  == NVIDIA and avoid the fatal error.

I've just tested that by changing
```cmake
  IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
      MESSAGE(FATAL_ERROR "Compiler ${KOKKOS_CXX_COMPILER_ID} should support ${FEATURE_NAME}, but CMake reports feature not supported")
  ENDIF()
```

into

```cmake
  IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
    IF (KOKKOS_CXX_COMPILER_ID STREQUAL "NVIDIA")
      MESSAGE(STATUS "nvcc_wrapper does not support TARGET_COMPILE_FEATURES")
      GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
    ELSE()
      MESSAGE(FATAL_ERROR "Compiler ${KOKKOS_CXX_COMPILER_ID} should support ${FEATURE_NAME}, but CMake reports feature not supported")
    ENDIF()
  ENDIF()
```

in cmake/kokkos_test_cxx_std.cmake in current develop, the CUDA build works again.

I guess there better ways to fix this problem.

Thanks.

**Comment by jjwilke** [comment]
This is a known `nvcc_wrapper` issue where it fails compiler feature tests. There are a couple more fixes that need to go in to `nvcc_wrapper` for downstream projects.

**Comment by masterleinad** [comment]
Fixed in #2474.

**Comment by pkestene** [comment]
@jjwilke @masterleinad Thanks !
